app-crypt/nitrocli/nitrocli-0.1.3.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CRATES="
+cc-1.0.25
+hid-0.4.1
+hidapi-sys-0.1.4
+libc-0.2.45
+nitrocli-0.1.3
+pkg-config-0.3.14
+"
+
+inherit cargo
+
+DESCRIPTION="A command line tool for interacting with the Nitrokey Storage"
+HOMEPAGE="https://github.com/d-e-s-o/nitrocli"
+SRC_URI="$(cargo_crate_uris ${CRATES})"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+RESTRICT="mirror"
+
+DEPEND="
+	>=dev-lang/rust-1.31.0
+"
+# We require gnupg for /usr/bin/gpg-connect-agent.
+RDEPEND="
+	${DEPEND}
+	app-crypt/gnupg
+	dev-libs/hidapi
+"
+
+src_install() {
+	cargo install -j $(makeopts_jobs) --path=. --root="${D}/usr" $(usex debug --debug "") \
+		|| die "cargo install failed"
+	rm -f "${D}/usr/.crates.toml"
+
+	doman "${S}/doc/nitrocli.1"
```

Comment:
you may want to call `einstalldocs` do install the documentation.

Reply:
> you can omit calling cargo_src_install altogether and just install binary with dobin
> it's located in target/release/nitrocli

That won't honor environment variables. Once the eclass is fixed I want to switch back to using `cargo_src_install`. So there is no point and diverging needlessly from my point of view.

> you may want to call einstalldocs to install the documentation.

It does not seem to handle man pages correctly.

Reply:
`einstalldocs` is handling docs, not man, you should keep the `doman`.

Reply:
> That won't honor environment variables. 

which variables?
You'll be able to switch to `cargo_src_install`, but for now keep it simple really. No need to call external build system if you can just `dobin` a single file.
if next release gets fixed to work with `cargo_src_install` or eclass changes then this ebuild will still work and new one will use `cargo_src_install` as it should.

Reply:
`cargo` honors a set of [environment variables](https://doc.rust-lang.org/cargo/reference/environment-variables.html), one of which is `CARGO_TARGET_DIR`. If this variable is set then the binary -- as compiled by `cargo` through `src_compile` -- won't be located at `target/release/nitrocli`. Last I checked you did not filter out those variables, so they would very well be in effect. And even if not, this is an inconsistency that is completely non-obvious.